src/GitHub.App/ViewModels/PullRequestCreationViewModel.cs

```diff
@@ -96,68 +119,93 @@ public class PullRequestCreationViewModel : BaseViewModel, IPullRequestCreationV
         {
-            initialized = false;
             base.Initialize(data);
 
-            this.Title = null;
-            this.Description = service.GetPullRequestTemplate(activeRepo) ?? string.Empty;
-
-            repositoryHost.ModelService.GetBranches(activeRepo)
-                            .ToReadOnlyList()
-                            .ObserveOn(RxApp.MainThreadScheduler)
-                            .Subscribe(x =>
-                            {
-                                Branches = x;
-                                initialized = true;
-                                initializationComplete.OnNext(Unit.Default);
-                            });
+            Initialized = false;
+
+            githubObs.SelectMany(r =>
+            {
+                var branches = Observable.Empty<IBranch>();
+                if (r.IsFork)
+                {
+                    branches = repositoryHost.ModelService.GetBranches(r.Parent).Select(x =>
+                    {
+                        x.DisplayName = x.Id;
+                        return x;
+                    });
+                }
+                return branches.Concat(repositoryHost.ModelService.GetBranches(r));
+            })
+            .ToList()
+            .ObserveOn(RxApp.MainThreadScheduler)
+            .Subscribe(x =>
+            {
+                BranchesList = new List<IBranch>(x);
+                Initialized = true;
+            });
+        }
+
+        public IRepositoryModel GitHubRepository { get { return githubRepository?.Value; } }
+        bool IsExecuting { get { return isExecuting.Value; } }
+
+        bool initialized;
+        bool Initialized
+        {
+            get { return initialized; }
+            set { this.RaiseAndSetIfChanged(ref initialized, value); }
         }
 
         IBranch sourceBranch;
-        [AllowNull]
         public IBranch SourceBranch
         {
-            [return: AllowNull]
             get { return sourceBranch; }
             set { this.RaiseAndSetIfChanged(ref sourceBranch, value); }
         }
 
         IBranch targetBranch;
-        [AllowNull]
         public IBranch TargetBranch
         {
-            [return: AllowNull]
             get { return targetBranch; }
             set { this.RaiseAndSetIfChanged(ref targetBranch, value); }
         }
 
-        IReadOnlyList<IBranch> branches;
-        public IReadOnlyList<IBranch> Branches
+        List<IBranch> branchesList;
+        List<IBranch> BranchesList
         {
-            [return: AllowNull]
-            get { return branches; }
-            set { this.RaiseAndSetIfChanged(ref branches, value); }
+            get { return branchesList; }
+            set
+            {
+                branchesList = value;
+                this.RaisePropertyChanged();
+                this.RaisePropertyChanged(nameof(Branches));
+            }
         }
 
+        public IReadOnlyList<IBranch> Branches => branchesList;
```

Comment:
Why are both `Branches` and `BranchesList` needed? The branches are never mutated after they're set so can't there be just `Branches`?

```
        IReadOnlyList<IBranch> branches;
        public IReadOnlyList<IBranch> Branches
        {
            get { return branches; }
            private set
            {
                branches = value;
                this.RaisePropertyChanged();
            }
        }
```